packages/calcite-components/src/components/tree/tree.tsx

```diff
@@ -244,72 +245,103 @@ export class Tree {
     event.stopPropagation();
   }
 
-  @Listen("keydown")
-  keyDownHandler(event: KeyboardEvent): void {
-    const root = this.el.closest("calcite-tree:not([child])") as HTMLCalciteTreeElement;
+  private keyDownHandler = (event: KeyboardEvent): void => {
+    if (this.child) {
+      return;
+    }
+
+    const root = this.el;
     const target = event.target as HTMLCalciteTreeItemElement;
 
-    if (!(root === this.el && target.tagName === "CALCITE-TREE-ITEM" && this.el.contains(target))) {
+    if (
+      !(isTreeItem(target) && this.el.contains(target)) ||
+      (event.key !== "ArrowRight" &&
```

Comment:
nitpick: Might be nicer as `const allowedKeys = ["ArrowDown", "ArrowLeft",...]`

Then `if(!allowedKeys.includes(event.key)) { return; }`